examples/pix2pose/eval.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+from scipy import spatial
+from scipy.spatial import distance
+from paz.backend.groups.quaternion import quaternion_to_rotation_matrix
+
+
+def get_translation(pose6D):
+    """It outputs the predicted Translation from the detected object.
+
+      # Arguments
+          pose6D: The 6d pose
+
+      # Returns
+          Return Translation vector
+    """
+    translation = np.array(pose6D.translation)  
+    return translation
+
+
+def get_quaternion(pose6D):
+    """It outputs the predicted quaternion from the detected object.
+
+      # Arguments
+          pose6D: The 6d pose
+
+      # Returns
+          Return Quaternion vector
+      """
+    quaternion = np.array(pose6D.quaternion)
+    return quaternion
+
+
+def transform_points3D(points3D, rotation, translation):
+    """Transforms the object points
+
+      # Arguments
+          3D_points: nx3 ndarray with 3D model points.
+          rotaion: Rotation matrix
+          translation: Translation vector
+
+      # Returns
+          Transformed model
+      """
+    transformed_points = rotation @ points3D.T + translation.reshape((3, 1))
+    return transformed_points.T
+
+
+def compute_ADD(true_pose, pred_pose, points3D):
+    """Calculate The ADD error.
+
+      # Arguments
+          true_pose: Real pose
+          pred_pose: Predicted pose
+          mesh_pts: nx3 ndarray with 3D model points.
+                   
+      # Returns
+          Return ADD error
+      """
+    pred_rotation = quaternion_to_rotation_matrix(pred_pose.quaternion)
+    pred_translation = pred_pose.translation
+    transformed_points_pred = transform_points3D(points3D, pred_rotation, pred_translation)
+    true_rotation = quaternion_to_rotation_matrix(true_pose.quaternion)
+    true_translation = pred_pose.translation
```

Comment:
Maybe we can change order to make it read faster?
```
pred_rotation = quaternion_to_rotation_matrix(pred_pose.quaternion)
true_rotation = quaternion_to_rotation_matrix(true_pose.quaternion)
pred_transformed_points = transform_points3D(points3D, pred_rotation, pred.translation)
true_transformed_points = transform_points3D(points3D, true_rotation, true.translation)
error = np.linalg.norm(pred_transformed_points - true_transformed_points, axis=1)
return error.mean()
```

Reply:
resolved